commit 66dfec8053be
Author: Daniel Minor <dminor@mozilla.com>
Date:   Wed Jun 20 21:18:50 2018 -0400

    Bug 1458048 - Cherry pick usersctp rev 8789a6da02e2c7c03522bc6f275b302f6ef977fe; r=drno, a=RyanVM
    
    --HG--
    extra : source : 492697e43fb4abf3cca014249bc35797f3312236
    extra : amend_source : 2a960ee15610dd344ceffd22dab70994cbf52198
---
 netwerk/sctp/src/netinet/sctp_auth.c | 4 +++-
 netwerk/sctp/src/netinet/sctp_pcb.c  | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git netwerk/sctp/src/netinet/sctp_auth.c netwerk/sctp/src/netinet/sctp_auth.c
index 50432ad8a5954..ee5ca36ce37b1 100755
--- netwerk/sctp/src/netinet/sctp_auth.c
+++ netwerk/sctp/src/netinet/sctp_auth.c
@@ -1525,6 +1525,8 @@ sctp_auth_get_cookie_params(struct sctp_tcb *stcb, struct mbuf *m,
 	    if (p_random != NULL) {
 		keylen = sizeof(*p_random) + random_len;
 		bcopy(p_random, new_key->key, keylen);
+	    } else {
+		keylen = 0;
 	    }
 	    /* append in the AUTH chunks */
 	    if (chunks != NULL) {
diff --git netwerk/sctp/src/netinet/sctp_pcb.c netwerk/sctp/src/netinet/sctp_pcb.c
index 2970970250a2a..58c164f5039c7 100755
--- netwerk/sctp/src/netinet/sctp_pcb.c
+++ netwerk/sctp/src/netinet/sctp_pcb.c
@@ -7688,6 +7688,8 @@ sctp_load_addresses_from_init(struct sctp_tcb *stcb, struct mbuf *m,
 		if (p_random != NULL) {
 			keylen = sizeof(*p_random) + random_len;
 			bcopy(p_random, new_key->key, keylen);
+		} else {
+			keylen = 0;
 		}
 		/* append in the AUTH chunks */
 		if (chunks != NULL) {
